sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -812,6 +812,15 @@ Flux<ServiceBusReceivedMessage> peekMessages(int maxMessages, long sequenceNumbe
      *     <li>An {@link AmqpException} occurs that causes the receive link to stop.</li>
      * </ul>
      *
+     * <p>
+     * The client uses an amqp-link underneath to receive the messages; the client will transparently transition to
+     * a new amqp-link if the current one encounters a retriable error. When the client experiences a non-retriable error
```

Comment:
should retriable error and non-retrieable errors be links to definitions

Reply:
Thanks & good point, I checked this, and there is no purely language-agnostic documentation. The one from the _Service Team_ has some generic doc but is combined with the .NET types. We should chat about developing a generic doc and then specializing it for languages.

Until there is such a Uber doc, I think adding inline examples for non-retriable errors is a good idea, as that's what the user ends up auditing. I've added examples for common cases.

Reply:
The doc from _Service Team_ is [here ](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-exceptions)